modules/tinymce/src/core/main/ts/delete/InlineFormatDelete.ts

```diff
@@ -23,6 +23,8 @@ const getParentsUntil = (editor: Editor, pred: (elm: SugarElement<Node>) => bool
     (index) => parents.slice(0, index)
   );
 };
+const hasOnlyOneChild = (elm: SugarElement<Node>): boolean =>
+  Traverse.childNodesCount(elm) === 1;
 
```

Comment:
Let's revert this back to the original behaviour before https://github.com/tinymce/tinymce/pull/8300. This PR attempted to refactor the `hasOnlyOneChild` code into something that can be reused for `deleteRange`, by finding all parents _until_ a multi-child parent, which caused this bug.
```suggestion

const getParentInlines = (editor: Editor): SugarElement<Node>[] =>
  getParentsUntil(editor, (elm) => ElementType.isBlock(elm));

```

Reply:
I tried to do this and I ran into a couple of test-failures because of it. It seems like we to some degree use the slightly different behavior.
Because of this I won't revert to the original behavior as I don't think these tests are wrong.

Reply:
Tests doing their job! Good you added them.